xtext/org.icyphy.linguafranca/src/org/icyphy/generator/CGenerator.xtend

```diff
@@ -445,7 +445,7 @@ class CGenerator extends GeneratorBase {
             // If federated, append the federate name to the file name.
             // Only generate one output if there is no federation.
             if (!federate.isSingleton) {
-                topLevelName = baseFilename + '_' + federate.name
+                topLevelName = baseFilename + '_' + federate.name // FIXME: don't do this
```

Comment:
Not to use a class variable which after this federate-specific code gets changed back to its original value. It's confusing. I think `topLevelName` should disappear altogether because we have it in `FileConfig.name` where it is `final`. I believe I left a FIXME for that too.

Reply:
Clarified the `FIXME`